src/main/java/team/serenity/logic/commands/question/EditQnCommand.java

```diff
@@ -0,0 +1,198 @@
+
+        // Checks if the edited question exists in the question manager
+        if (!questionToEdit.equals(editedQuestion) && model.hasQuestion(editedQuestion)) {
+            throw new CommandException(MESSAGE_DUPLICATE_QUESTION);
+        }
+
+        // Checks if the edited question's group name exists in serenity
+        if (!questionToEdit.getGroupName().equals(editedQuestion.getGroupName())
+            && !model.hasGroupName(editedQuestion.getGroupName())) {
+            throw new CommandException(String.format(MESSAGE_GROUP_NOT_FOUND_FORMAT,
+                editedQuestion.getGroupName().groupName));
+        }
+
+        // Checks if the edited question's lesson name exists in the group and serenity
+        if (!questionToEdit.getLessonName().equals(editedQuestion.getLessonName())
+            && !model.ifTargetGroupHasLessonName(editedQuestion.getGroupName(), editedQuestion.getLessonName())) {
+            throw new CommandException(String.format(MESSAGE_LESSON_NOT_FOUND_FORMAT,
+                    editedQuestion.getLessonName().lessonName));
+        }
+
+        model.setQuestion(questionToEdit, editedQuestion);
+        model.updateFilteredQuestionList(Model.PREDICATE_SHOW_ALL_QUESTIONS);
+        // TODO: Ryan/Ruien to add the viewQn boolean to CommandResult constructor to switch to view question tab
+        return new CommandResult(String.format(MESSAGE_EDIT_QUESTION_SUCCESS, editedQuestion));
+    }
+
+    /**
+     * Creates and returns a {@code Question} with the details of {@code questionToEdit}
+     * edited with {@code editQuestionDescriptor}
+     */
+    private static Question createEditedQuestion(Question questionToEdit,
+                                                 EditQuestionDescriptor editQuestionDescriptor) {
+        assert questionToEdit != null; // the question to edit is taken from the filtered question list
+
+        GroupName updatedGroupName = editQuestionDescriptor.getGroupName().orElse(questionToEdit.getGroupName());
+        LessonName updatedLessonName = editQuestionDescriptor.getLessonName().orElse(questionToEdit.getLessonName());
+        Description updatedDescription = editQuestionDescriptor.getDescription()
+                .orElse(questionToEdit.getDescription());
+
+        return new Question(updatedGroupName, updatedLessonName, updatedDescription);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+            || (other instanceof EditQnCommand // instanceof handles nulls
+            && this.index.equals(((EditQnCommand) other).index)
+            && this.editQuestionDescriptor.equals(((EditQnCommand) other).editQuestionDescriptor));
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.index.hashCode(), this.editQuestionDescriptor.hashCode());
+    }
+
+    /**
+     * Stores the details to edit the fixed expense with. Each non-empty field value will replace the
```

Comment:
The javadoc here is inaccurate

Reply:
Will edit it, thanks for pointing out :)